src/librustc/mir/visit.rs

```diff
@@ -486,8 +486,12 @@ macro_rules! make_mir_visitor {
                         self.visit_operand(value, source_location);
                         self.visit_branch(block, resume);
                         drop.map(|t| self.visit_branch(block, t));
+
                     }
 
+                    TerminatorKind::FalseEdges { real_target, .. } => {
```

Comment:
I think you should visit the imaginary edges too, so that code that uses this to compute the CFG gets the "correct" approximation.

I don't think anyone actually listens on `visit_branch`, but it's still the right thing.